### PR TITLE
feat: Implement endpoint to get blogs by Id

### PR DESCRIPTION
**Pull request by Nifilat**
​

## Description

Added functionality for retrieving blog posts by id. This includes creating a new endpoint /api/v1/blogs/{id} that fetches
blog posts that match the id from the database. The endpoint returns a single blog post, containing details such as title, content, author, tags, etc.

​

## Related Issue (Link to issue ticket)

[#236 ](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/236)
​

## Motivation and Context

​This change is required to provide an endpoint for clients to retrieve a blog post by its id. It enhances the application's functionality by allowing users to view a single blog post.

## How Has This Been Tested?

The tests were carried out using pytest.
I created a mock blog and was able to fetch the details successfully. 
I also tested for a successful 404 response for IDs that do not exist

### Success Response:
Screenshot 2024-07-25 at 02 28 23

### Error Response:
Screenshot 2024-07-25 at 08 58 09

## Swagger Documentation
Screenshot 2024-07-24 at 22 08 13

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: api/v1/services/blog.py [hunk 1]
```diff
@@ -1,8 +1,71 @@
...
+# #     def fetch(self, blog_id: str):
+# #         '''Fetch a blog post by its ID'''
+# #         blog_post = self.db.query(Blog).filter(Blog.id == blog_id).first()
+# #         if not blog_post:
+# #             raise HTTPException(status_code=404, detail="Post not Found")
+# #         return blog_post
+
+# #     def update(self, blog_id: str, title: Optional[str] = None, content: Optional[str] = None, current_user: User = None):
+# #         '''Updates a blog post'''
+
```
Comment (Goketech) [inline api/v1/services/blog.py:68]: you commented all this out

fix this

**Review by Goketech — CHANGES_REQUESTED** [review]

File: api/v1/services/blog.py (hunk 1 above)
Comment (Nifilat) [reply api/v1/services/blog.py]: I've fixed it

File: tests/v1/test_blog_update.py [hunk 2]
```diff
@@ -9,44 +15,46 @@
 # Add the project root directory to the Python path
 sys.path.append(str(Path(__file__).resolve().parent.parent.parent))
 
-from main import app
-from api.utils.dependencies import get_db, get_current_user
-from api.v1.models.user import User
-from api.v1.models.blog import Blog
-from api.v1.schemas.blog import BlogRequest
-from fastapi.encoders import jsonable_encoder
```
Comment (Goketech) [inline tests/v1/test_blog_update.py:17]: you made changes to a different file?

if not, pull from dev branch and make sure your branch is up to date

**Review by Goketech — CHANGES_REQUESTED** [review]

File: tests/v1/test_blog_update.py (hunk 2 above)
Comment (Nifilat) [reply tests/v1/test_blog_update.py:17]: They were not changes, the test files kept giving an error so I wanted to know what was wrong

**Review by Goketech — CHANGES_REQUESTED** [review]
revert all changes in test_blog_update.py

pull from dev branch and make sure your pr is up to date

**Review by Goketech — DISMISSED** [review]

**Review by Goketech — APPROVED** [review]

**Review by joboy-dev — APPROVED** [review]

**Review by bolexs — APPROVED** [review]

**Pull request merged**